core/shared/src/main/scala/fs2/async/async.scala

```diff
@@ -134,44 +131,14 @@ package object async {
     * Begins asynchronous evaluation of `f` when the returned `F[Unit]` is
     * bound. Like `start` but is more efficient.
     */
-  def fork[F[_], A](f: F[A])(implicit F: Effect[F], ec: ExecutionContext): F[Unit] =
-    F.liftIO(F.runAsync(F.shift *> f) { _ =>
-      IO.unit
-    })
+  def fork[F[_], A](f: F[A])(implicit F: Concurrent[F], ec: ExecutionContext): F[Unit] =
+    F.start(Async.shift(ec) *> f).void
```

Comment:
Any reason for the `shift`? No opinions, but curious if it was needed to make things work.
Ideally we could get rid of `EC`s in the api

Reply:
Only to keep the old behavior -- unlike `fs2.async.start`, `Concurrent[F].start` does not introduce a thread shift.

Reply:
Ok, then I'd say it goes into the broader exploration of `ec`s and when and where we want to shift.